docs/plugin/interfaces/interfaces.md

```diff
@@ -0,0 +1,603 @@
+Thunder plugins are built around the concept of interfaces. An interface acts as a contract between the plugin and the outside world (this could be external client applications or other plugins). 
+
+In Thunder, plugins can expose their interfaces over two communication channels:
+
+* JSON-RPC ([https://www.jsonrpc.org/specification](https://www.jsonrpc.org/specification))
+* COM-RPC
+
+In older Thunder versions (<R4), it was common for plugins to define two interface files, one for each RPC format. In R4 and later versions, it is strongly recommended to use the code-generation tools included in Thunder and only write a single interface file. This makes it much easier to maintain and reduces the amount of boilerplate code that must be written in a plugin. This is the approach documented in this page. 
+
+Examples of existing Thunder interfaces can be found in the ThunderInterfaces repository: [https://github.com/rdkcentral/ThunderInterfaces/](https://github.com/rdkcentral/ThunderInterfaces/).
+
+## Designing a good Interface
+A good interface defines a set of methods that a plugin can support, without dictating anything about the actual code that will implement the interface. It is possible many plugins could implement that same interface if they provide overlapping functionality (for example an `IBrowser` interface could be implemented by different web browser engines).
+
+The interface provides a clear boundary between the code that invokes methods on the plugin and the plugin code that implements the functionality. Typically, interface definitions are stored in a separate repository to the plugin implementations to reflect this boundary.
+
+Well designed interfaces should also be easy to read and self explanatory. Methods names should be descriptive and have doxygen-style comments that describe their inputs and outputs. All methods should return an error code to indicate whether they completed successfully, and any data that should be returned by the method should be stored in an output parameter.
```

Comment:
"All methods should return an error code" don't we want to mention her that is is prefered to be Core::hresult ? Or something like: "All methods should return an error code (core::hresult)", maybe with reasoning so that the COMRPC errors can be reported in the upper part or something like that ?

Reply:
I cover this later on in the "Guidelines" section, don't think it needs adding here as well